### PR TITLE
Update joomla/string package to 1.2.2

### PR DESCRIPTION
**Pull request by mbabker**
Please see https://github.com/joomla-framework/string/compare/1.2.1...1.2.2 for changes in the String package.

What will be the most noticeable change here is that the issue addressed in #6405 is resolved.  Otherwise, the rest of the changes are generally architecture related changes (less else uses) that don't affect the code.

### CODE REVIEW

File: libraries/vendor/joomla/string/src/Inflector.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /**
  * Part of the Joomla Framework String Package
  *
- * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @copyright  Copyright (C) 2005 - 2013 Open Source Matters, Inc. All rights reserved.
```
Comment (810) [inline libraries/vendor/joomla/string/src/Inflector.php]: please change this back to 2015 also next file

Comment (mbabker) [inline libraries/vendor/joomla/string/src/Inflector.php]: The copyrights need to be corrected in the upstream repo.  We shouldn't be altering anything in the libraries/vendor folder for any reason.

**Comment by joomla-agency** [comment]
How can i test this patch? Test instructions missing?!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6416">issues.joomla.org/joomla-cms/6416</a>.</sub>

**Comment by wilsonge** [comment]
We have very good unit test coverage for this class and as it's just updating a piece of upstream code I'm merging this on review

**Pull request merged**